File: lib/src/em_sdk_method.dart
class EMSDKMethod {
  /// EMClient methods
  static const String init = 'init';
  static const String createAccount = 'createAccount';
  static const String login = 'login';
  static const String logout = 'logout';
  static const String changeAppKey = 'changeAppKey';
  static const String isLoggedInBefore = 'isLoggedInBefore';
  static const String setNickname = 'setNickname';
  static const String uploadLog = 'uploadLog';
  static const String compressLogs = 'compressLogs';
  static const String kickDevice = 'kickDevice';
  static const String kickAllDevices = 'kickAllDevices';
  static const String currentUser =  'currentUser';
  static const String getLoggedInDevicesFromServer = 'getLoggedInDevicesFromServer';

  /// EMClient listener
  static const String onMultiDeviceEvent = 'onMultiDeviceEvent';
  static const String onConnected = 'onConnected';
  static const String onDisconnected = 'onDisconnected';

  /// EMContactManager methods
  static const String addContact = 'addContact';
  static const String deleteContact = 'deleteContact';
  static const String getAllContactsFromServer = 'getAllContactsFromServer';
  static const String addUserToBlackList = 'addUserToBlackList';
  static const String removeUserFromBlackList = 'removeUserFromBlackList';
  static const String getBlackListFromServer = 'getBlackListFromServer';
  static const String acceptInvitation = 'acceptInvitation';
  static const String declineInvitation = 'declineInvitation';
  static const String getSelfIdsOnOtherPlatform = 'getSelfIdsOnOtherPlatform';

  /// EMContactManager listener
  static const String onContactChanged = 'onContactChanged';

  /// EMChatManager methods
  static const String sendMessage = 'sendMessage';
  static const String ackMessageRead = 'ackMessageRead';
  static const String recallMessage = 'recallMessage';
  static const String getConversation = 'getConversation';
  static const String markAllChatMsgAsRead = 'markAllChatMsgAsRead';
  static const String getUnreadMessageCount = 'getUnreadMessageCount';
  static const String updateChatMessage = 'updateChatMessage';
  static const String downloadAttachment = 'downloadAttachment';
  static const String downloadThumbnail = 'downloadThumbnail';
  static const String importMessages = 'importMessages';
  static const String loadAllConversations = 'loadAllConversations';
  static const String deleteConversation = 'deleteConversation';
  static const String fetchHistoryMessages = 'fetchHistoryMessages';
  static const String searchChatMsgFromDB = 'searchChatMsgFromDB';
  static const String getMessage = 'getMessage';

  /// EMChatManager listener
  static const String onMessagesReceived = 'onMessagesReceived';
  static const String onCmdMessagesReceived = 'onCmdMessagesReceived';
  static const String onMessagesRead = 'onMessagesRead';
  static const String onMessagesDelivered = 'onMessagesDelivered';
  static const String onMessagesRecalled = 'onMessagesRecalled';
  static const String onMessageChanged = 'onMessageChanged';

  static const String onConversationUpdate = 'onConversationUpdate';

  /// EMMessage listener
  static const String onMessageProgressUpdate = 'onMessageProgressUpdate';
  static const String onMessageError = 'onMessageError';
  static const String onMessageSuccess = 'onMessageSuccess';
  static const String onMessageReadAck = 'onMessageReadAck';
  static const String onMessageDeliveryAck = 'onMessageDeliveryAck';
  static const String onMessageStatusChanged = 'onMessageStatusChanged';

  /// EMConversation
  static const String getUnreadMsgCount = 'getUnreadMsgCount';
  static const String markAllMessagesAsRead = 'markAllMessagesAsRead';
  static const String markMessageAsRead = 'markMessageAsRead';
  static const String removeMessage = 'removeMessage';
  static const String getLatestMessage = 'getLatestMessage';
  static const String getLatestMessageFromOthers = 'getLatestMessageFromOthers';
  static const String clearAllMessages = 'clearAllMessages';
  static const String insertMessage = 'insertMessage';
  static const String appendMessage = 'appendMessage';
  static const String updateConversationMessage = 'updateConversationMessage';

  // 根据消息id获取消息
  static const String loadMsgWithId = 'loadMsgWithId';
  // 根据起始消息id获取消息
  static const String loadMsgWithStartId = 'loadMsgWithStartId';
  // 根据关键字获取消息
  static const String loadMsgWithKeywords = 'loadMsgWithKeywords';
  // 根据消息类型获取消息
  static const String loadMsgWithMsgType = 'loadMsgWithMsgType';
  // 通过时间获取消息
  static const String loadMsgWithTime = 'loadMsgWithTime';

  /// EMChatRoomManager methods
  static const String joinChatRoom = 'joinChatRoom';
  static const String leaveChatRoom = 'leaveChatRoom';
  static const String fetchPublicChatRoomsFromServer = 'fetchPublicChatRoomsFromServer';
  static const String fetchChatRoomInfoFromServer = 'fetchChatRoomInfoFromServer';
  static const String getChatRoom = 'getChatRoom';
  static const String getAllChatRooms = 'getAllChatRooms';
  static const String createChatRoom = 'createChatRoom';
  static const String destroyChatRoom = 'destroyChatRoom';
  static const String changeChatRoomSubject = 'changeChatRoomSubject';
  static const String changeChatRoomDescription = 'changeChatRoomDescription';
  static const String fetchChatRoomMembers = 'fetchChatRoomMembers';
  static const String muteChatRoomMembers = 'muteChatRoomMembers';
  static const String unMuteChatRoomMembers = 'unMuteChatRoomMembers';
  static const String changeChatRoomOwner = 'changeChatRoomOwner';
  static const String addChatRoomAdmin = 'addChatRoomAdmin';
  static const String removeChatRoomAdmin = 'removeChatRoomAdmin';
  static const String fetchChatRoomMuteList = 'fetchChatRoomMuteList';
  static const String removeChatRoomMembers = 'removeChatRoomMembers';
  static const String blockChatRoomMembers = 'blockChatRoomMembers';
  static const String unBlockChatRoomMembers = 'unBlockChatRoomMembers';
  static const String fetchChatRoomBlackList = 'fetchChatRoomBlackList';
  static const String updateChatRoomAnnouncement = 'updateChatRoomAnnouncement';
  static const String fetchChatRoomAnnouncement = 'fetchChatRoomAnnouncement';


  /// EMChatRoomManagerListener
  static const String chatRoomChange = 'onChatRoomChanged';

  /// EMGroupManager
  static const String getGroupWithId = 'getGroupWithId';
  static const String getJoinedGroups = 'getJoinedGroups';
  static const String getGroupsWithoutPushNotification = 'getGroupsWithoutPushNotification';
  static const String getJoinedGroupsFromServer = 'getJoinedGroupsFromServer';
  static const String getPublicGroupsFromServer = 'getPublicGroupsFromServer';
  static const String createGroup = 'createGroup';
  static const String getGroupSpecificationFromServer = 'getGroupSpecificationFromServer';
  static const String getGroupMemberListFromServer = 'getGroupMemberListFromServer';
  static const String getGroupBlacklistFromServer = 'getGroupBlacklistFromServer';
  static const String getGroupMuteListFromServer = 'getGroupMuteListFromServer';
  static const String getGroupWhiteListFromServer = 'getGroupWhiteListFromServer';
  static const String isMemberInWhiteListFromServer = 'isMemberInWhiteListFromServer';
  static const String getGroupFileListFromServer = 'getGroupFileList';
  static const String getGroupAnnouncementFromServer = 'getGroupAnnouncementFromServer';
  static const String addMembers = 'addMembers';
  static const String removeMembers = 'removeMembers';
  static const String blockMembers = 'blockMembers';
  static const String unblockMembers = 'unblockMembers';
  static const String updateGroupSubject = 'updateGroupSubject';
  static const String updateDescription = 'updateDescription';
  static const String leaveGroup = 'leaveGroup';
  static const String destroyGroup = 'destroyGroup';
  static const String blockGroup = 'blockGroup';
  static const String unblockGroup = 'unblockGroup';
  static const String updateGroupOwner = 'updateGroupOwner';
  static const String addAdmin = 'addAdmin';
  static const String removeAdmin = 'removeAdmin';
  static const String muteMembers = 'muteMembers';
  static const String unMuteMembers = 'unMuteMembers';
  static const String muteAllMembers = 'muteAllMembers';
  static const String unMuteAllMembers = 'unMuteAllMembers';
  static const String addWhiteList = 'addWhiteList';
  static const String removeWhiteList = 'removeWhiteList';
  static const String uploadGroupSharedFile = 'uploadGroupSharedFile';
  static const String downloadGroupSharedFile = 'downloadGroupSharedFile';
  static const String removeGroupSharedFile = 'removeGroupSharedFile';
  static const String updateGroupAnnouncement = 'updateGroupAnnouncement';
  static const String updateGroupExt = 'updateGroupExt';
  static const String joinPublicGroup = 'joinPublicGroup';
  static const String requestToJoinPublicGroup = 'requestToJoinPublicGroup';
  static const String acceptJoinApplication = 'acceptJoinApplication';
  static const String declineJoinApplication = 'declineJoinApplication';
  static const String acceptInvitationFromGroup = 'acceptInvitationFromGroup';
  static const String declineInvitationFromGroup = 'declineInvitationFromGroup';
  static const String ignoreGroupPush = 'ignoreGroupPush';

  /// EMGroupManagerListener
  static const String onGroupChanged = 'onGroupChanged';

  /// EMPushManager
  static const String getImPushConfigs = 'getImPushConfigs';
  static const String getImPushConfigsFromServer = 'getImPushConfigsFromServer';
  static const String updatePushNickname = 'updatePushNickname';

  /// ImPushConfigs
  static const String imPushNoDisturb = 'imPushNoDisturb';
  static const String updateImPushStyle = 'updateImPushStyle';
  static const String updateGroupPushService = 'updateGroupPushService';
  static const String getNoDisturbGroups = 'getNoDisturbGroups';

  /// EMCallManager
  static const String setCallOptions = 'setCallOptions';
  static const String getCallOptions = 'getCallOptions';


  static const String makeCall = 'makeCall';
  static const String answerCall = 'answerCall';
  static const String rejectCall = 'rejectCall';
  static const String endCall = 'endCall';

  static const String enableVoiceTransfer = 'enableVoiceTransfer';
  static const String enableVideoTransfer = 'enableVideoTransfer';

  static const String muteRemoteAudio = 'muteRemoteAudio';
  static const String muteRemoteVideo = 'muteRemoteVideo';

  static const String switchCamera = 'switchCamera';
  static const String setSurfaceView = 'setSurfaceView';
  static const String releaseView = 'releaseView';

  /// EMCallManager Listener
  static const String onCallReceived = 'onCallReceived';
  static const String onCallAccepted = 'onCallAccepted';
  static const String onCallRejected = 'onCallRejected';
  static const String onCallHangup = 'onCallHangup';
  static const String onCallBusy = 'onCallBusy';
  static const String onCallVideoPause = 'onCallVideoPause';
  static const String onCallVideoResume = 'onCallVideoResume';
  static const String onCallVoicePause = 'onCallVoicePause';
  static const String onCallVoiceResume = 'onCallVoiceResume';
  static const String onCallNetworkUnStable = 'onCallNetworkUnStable';
  static const String onCallNetworkNormal = 'onCallNetworkNormal';
  static const String onCallNetworkDisconnect = 'onCallNetworkDisconnect';


  /// EMConferenceManager
  static const String setConferenceAppKey = 'setConferenceAppKey';
  static const String conferenceHasExists = 'conferenceHasExists';
  static const String joinConference = 'joinConference';
  static const String joinRoom = 'joinRoom';

  static const String createWhiteboardRoom = 'createWhiteboardRoom';
  static const String destroyWhiteboardRoom = 'destroyWhiteboardRoom';
  static const String joinWhiteboardRoom = 'joinWhiteboardRoom';

  /// EMConference
  static const String publishConference = 'publishConference';
  static const String unPublishConference = 'unPublishConference';
  static const String subscribeConference = 'subscribeConference';
  static const String unSubscribeConference = 'unSubscribeConference';

  static const String changeMemberRoleWithMemberName = 'changeMemberRoleWithMemberName';
  static const String kickConferenceMember = 'kickConferenceMember';
  static const String destroyConference = 'destroyConference';
  static const String leaveConference = 'leaveConference';

  static const String startMonitorSpeaker = 'startMonitorSpeaker';
  static const String stopMonitorSpeaker = 'stopMonitorSpeaker';

  static const String requestTobeConferenceSpeaker = 'requestTobeConferenceSpeaker';
  static const String requestTobeConferenceAdmin = 'requestTobeConferenceAdmin';
  static const String muteConferenceMember = 'muteConferenceMember';

  static const String responseReqSpeaker = 'responseReqSpeaker';
  static const String responseReqAdmin = 'responseReqAdmin';
  static const String updateConferenceWithSwitchCamera = 'updateConferenceWithSwitchCamera';
  static const String updateConferenceMute = 'updateConferenceMute';
  static const String updateConferenceVideo = 'updateConferenceVideo';

  static const String updateRemoteView = 'updateRemoteView';
  static const String updateMaxVideoKbps = 'updateMaxVideoKbps';

  static const String setConferenceAttribute = 'setConferenceAttribute';
  static const String deleteAttributeWithKey = 'deleteAttributeWithKey';

  static const String muteConferenceRemoteAudio = 'muteConferenceRemoteAudio';
  static const String muteConferenceRemoteVideo = 'muteConferenceRemoteVideo';
  static const String muteConferenceAll = 'muteConferenceAll';

  static const String addVideoWatermark = 'addVideoWatermark';
  static const String clearVideoWatermark = 'clearVideoWatermark';
}

File: android/src/main/java/com/easemob/im_flutter_sdk/EMSDKMethod.java
package com.easemob.im_flutter_sdk;

public class EMSDKMethod {
    /// EMClient methods
    static final String init = "init";
    static final String createAccount = "createAccount";
    static final String login = "login";
    static final String loginWithAgoraToken = "loginWithAgoraToken";
    static final String renewToken = "renewToken";
    static final String logout = "logout";
    static final String changeAppKey = "changeAppKey";

    static final String uploadLog = "uploadLog";
    static final String compressLogs = "compressLogs";
    static final String kickDevice = "kickDevice";
    static final String kickAllDevices = "kickAllDevices";
    static final String getLoggedInDevicesFromServer = "getLoggedInDevicesFromServer";

    static final String getToken = "getToken";
    static final String getCurrentUser = "getCurrentUser";
    static final String isLoggedInBefore = "isLoggedInBefore";
    static final String isConnected = "isConnected";

    static final String onConnected = "onConnected";
    static final String onDisconnected = "onDisconnected";
    static final String onUserDidLoginFromOtherDevice = "onUserDidLoginFromOtherDevice";
    static final String onUserDidRemoveFromServer = "onUserDidRemoveFromServer";
    static final String onUserDidForbidByServer = "onUserDidForbidByServer";
    static final String onUserDidChangePassword = "onUserDidChangePassword";
    static final String onUserDidLoginTooManyDevice = "onUserDidLoginTooManyDevice";
    static final String onUserKickedByOtherDevice = "onUserKickedByOtherDevice";
    static final String onUserAuthenticationFailed = "onUserAuthenticationFailed";
    static final String onAppActiveNumberReachLimit = "onAppActiveNumberReachLimit";
    static final String onMultiDeviceGroupEvent = "onMultiDeviceGroupEvent";
    static final String onMultiDeviceContactEvent = "onMultiDeviceContactEvent";
    static final String onMultiDeviceThreadEvent = "onMultiDeviceThreadEvent";
    static final String onMultiDeviceRemoveMessagesEvent = "onMultiDeviceRemoveMessagesEvent";
    static final String onMultiDevicesConversationEvent = "onMultiDevicesConversationEvent";
    static final String onSendDataToFlutter = "onSendDataToFlutter";
    static final String onTokenWillExpire = "onTokenWillExpire";
    static final String onTokenDidExpire = "onTokenDidExpire";

    /// EMContactManager methods
    static final String addContact = "addContact";
    static final String deleteContact = "deleteContact";
    static final String getAllContactsFromServer = "getAllContactsFromServer";
    static final String getAllContactsFromDB = "getAllContactsFromDB";
    static final String addUserToBlockList = "addUserToBlockList";
    static final String removeUserFromBlockList = "removeUserFromBlockList";
    static final String getBlockListFromServer = "getBlockListFromServer";
    static final String getBlockListFromDB = "getBlockListFromDB";
    static final String acceptInvitation = "acceptInvitation";
    static final String declineInvitation = "declineInvitation";
    static final String getSelfIdsOnOtherPlatform = "getSelfIdsOnOtherPlatform";


    static final String onContactChanged = "onContactChanged";

    /// EMChatManager methods
    static final String sendMessage = "sendMessage";
    static final String resendMessage = "resendMessage";
    static final String ackMessageRead = "ackMessageRead";
    static final String ackGroupMessageRead = "ackGroupMessageRead";
    static final String ackConversationRead = "ackConversationRead";
    static final String recallMessage = "recallMessage";
    static final String getConversation = "getConversation";
    static final String getThreadConversation = "getThreadConversation";
    static final String markAllChatMsgAsRead = "markAllChatMsgAsRead";
    static final String getUnreadMessageCount = "getUnreadMessageCount";
    static final String updateChatMessage = "updateChatMessage";
    static final String downloadAttachment = "downloadAttachment";
    static final String downloadThumbnail = "downloadThumbnail";
    static final String importMessages = "importMessages";
    static final String loadAllConversations = "loadAllConversations";
    static final String getConversationsFromServer = "getConversationsFromServer";
    static final String deleteConversation = "deleteConversation";
    static final String fetchHistoryMessages = "fetchHistoryMessages";
    static final String fetchHistoryMessagesByOptions = "fetchHistoryMessagesByOptions";
    static final String searchChatMsgFromDB = "searchChatMsgFromDB";
    static final String getMessage = "getMessage";
    static final String asyncFetchGroupAcks = "asyncFetchGroupAcks";
    static final String deleteRemoteConversation = "deleteRemoteConversation";
    static final String deleteMessagesBeforeTimestamp = "deleteMessagesBeforeTimestamp";

    static final String translateMessage = "translateMessage";
    static final String fetchSupportedLanguages = "fetchSupportLanguages";

    static final String addReaction = "addReaction";
    static final String removeReaction = "removeReaction";
    static final String fetchReactionList = "fetchReactionList";
    static final String fetchReactionDetail = "fetchReactionDetail";
    static final String reportMessage = "reportMessage";
    static final String fetchConversationsFromServerWithPage = "fetchConversationsFromServerWithPage";
    static final String removeMessagesFromServerWithMsgIds = "removeMessagesFromServerWithMsgIds";
    static final String removeMessagesFromServerWithTs = "removeMessagesFromServerWithTs";

    static final String getConversationsFromServerWithCursor = "getConversationsFromServerWithCursor";
    static final String getPinnedConversationsFromServerWithCursor = "getPinnedConversationsFromServerWithCursor";
    static final String pinConversation = "pinConversation";
    static final String modifyMessage = "modifyMessage";
    static final String downloadAndParseCombineMessage = "downloadAndParseCombineMessage";
    /// EMChatManager listener
    static final String onMessagesReceived = "onMessagesReceived";
    static final String onCmdMessagesReceived = "onCmdMessagesReceived";
    static final String onMessagesRead = "onMessagesRead";
    static final String onGroupMessageRead = "onGroupMessageRead";
    static final String onReadAckForGroupMessageUpdated = "onReadAckForGroupMessageUpdated";
    static final String onMessagesDelivered = "onMessagesDelivered";
    static final String onMessagesRecalled = "onMessagesRecalled";

    static final String onConversationUpdate = "onConversationUpdate";
    static final String onConversationHasRead = "onConversationHasRead";

    static final String onMessageReactionDidChange = "messageReactionDidChange";
    static final String onMessageContentChanged = "onMessageContentChanged";


    /// EMMessage listener
    static final String onMessageProgressUpdate = "onMessageProgressUpdate";
    static final String onMessageError = "onMessageError";
    static final String onMessageSuccess = "onMessageSuccess";
    static final String onMessageReadAck = "onMessageReadAck";
    static final String onMessageDeliveryAck = "onMessageDeliveryAck";

    /// EMConversation
    static final String getUnreadMsgCount = "getUnreadMsgCount";
    static final String markAllMessagesAsRead = "markAllMessagesAsRead";
    static final String markMessageAsRead = "markMessageAsRead";
    static final String syncConversationExt = "syncConversationExt";
    static final String removeMessage = "removeMessage";
    static final String getLatestMessage = "getLatestMessage";
    static final String getLatestMessageFromOthers = "getLatestMessageFromOthers";
    static final String clearAllMessages = "clearAllMessages";

    static final String deleteMessagesWithTs = "deleteMessagesWithTs";
    static final String insertMessage = "insertMessage";
    static final String appendMessage = "appendMessage";
    static final String updateConversationMessage = "updateConversationMessage";
    static final String loadMsgWithId = "loadMsgWithId";
    static final String loadMsgWithStartId = "loadMsgWithStartId";
    static final String loadMsgWithKeywords = "loadMsgWithKeywords";
    static final String loadMsgWithMsgType = "loadMsgWithMsgType";
    static final String loadMsgWithTime = "loadMsgWithTime";
    static final String messageCount = "messageCount";
    static final String removeMsgFromServerWithMsgList = "removeMsgFromServerWithMsgList";
    static final String removeMsgFromServerWithTimeStamp = "removeMsgFromServerWithTimeStamp";

    // EMMessage method
    static final String getReactionList = "getReactionList";
    static final String groupAckCount = "groupAckCount";
    static final String getChatThread = "chatThread";

    // EMChatRoomManager
    static final String joinChatRoom = "joinChatRoom";
    static final String leaveChatRoom = "leaveChatRoom";
    static final String fetchPublicChatRoomsFromServer = "fetchPublicChatRoomsFromServer";
    static final String fetchChatRoomInfoFromServer = "fetchChatRoomInfoFromServer";
    static final String getChatRoom = "getChatRoom";
    static final String getAllChatRooms = "getAllChatRooms";
    static final String createChatRoom = "createChatRoom";
    static final String destroyChatRoom = "destroyChatRoom";
    static final String changeChatRoomSubject = "changeChatRoomSubject";
    static final String changeChatRoomDescription = "changeChatRoomDescription";
    static final String fetchChatRoomMembers = "fetchChatRoomMembers";
    static final String muteChatRoomMembers = "muteChatRoomMembers";
    static final String unMuteChatRoomMembers = "unMuteChatRoomMembers";
    static final String changeChatRoomOwner = "changeChatRoomOwner";
    static final String addChatRoomAdmin = "addChatRoomAdmin";
    static final String removeChatRoomAdmin = "removeChatRoomAdmin";
    static final String fetchChatRoomMuteList = "fetchChatRoomMuteList";
    static final String removeChatRoomMembers = "removeChatRoomMembers";
    static final String blockChatRoomMembers = "blockChatRoomMembers";
    static final String unBlockChatRoomMembers = "unBlockChatRoomMembers";
    static final String fetchChatRoomBlockList = "fetchChatRoomBlockList";
    static final String updateChatRoomAnnouncement = "updateChatRoomAnnouncement";
    static final String fetchChatRoomAnnouncement = "fetchChatRoomAnnouncement";

    static final String addMembersToChatRoomWhiteList = "addMembersToChatRoomWhiteList";
    static final String removeMembersFromChatRoomWhiteList = "removeMembersFromChatRoomWhiteList";
    static final String fetchChatRoomWhiteListFromServer = "fetchChatRoomWhiteListFromServer";
    static final String isMemberInChatRoomWhiteListFromServer = "isMemberInChatRoomWhiteListFromServer";

    static final String muteAllChatRoomMembers = "muteAllChatRoomMembers";
    static final String unMuteAllChatRoomMembers = "unMuteAllChatRoomMembers";
    static final String fetchChatRoomAttributes = "fetchChatRoomAttributes";
    static final String setChatRoomAttributes = "setChatRoomAttributes";
    static final String removeChatRoomAttributes = "removeChatRoomAttributes";

    // EMChatRoomManagerListener
    static final String chatRoomChange = "onChatRoomChanged";

    /// EMGroupManager
    static final String getGroupWithId = "getGroupWithId";
    static final String getJoinedGroups = "getJoinedGroups";
    static final String getJoinedGroupsFromServer = "getJoinedGroupsFromServer";
    static final String getPublicGroupsFromServer = "getPublicGroupsFromServer";
    static final String createGroup = "createGroup";
    static final String getGroupSpecificationFromServer = "getGroupSpecificationFromServer";
    static final String getGroupMemberListFromServer = "getGroupMemberListFromServer";
    static final String getGroupBlockListFromServer = "getGroupBlockListFromServer";
    static final String getGroupMuteListFromServer = "getGroupMuteListFromServer";
    static final String getGroupWhiteListFromServer = "getGroupWhiteListFromServer";
    static final String isMemberInWhiteListFromServer = "isMemberInWhiteListFromServer";
    static final String getGroupFileListFromServer = "getGroupFileListFromServer";
    static final String getGroupAnnouncementFromServer = "getGroupAnnouncementFromServer";
    static final String addMembers = "addMembers";
    static final String inviterUser = "inviterUser";
    static final String removeMembers = "removeMembers";
    static final String blockMembers = "blockMembers";
    static final String unblockMembers = "unblockMembers";
    static final String updateGroupSubject = "updateGroupSubject";
    static final String updateDescription = "updateDescription";
    static final String leaveGroup = "leaveGroup";
    static final String destroyGroup = "destroyGroup";
    static final String blockGroup = "blockGroup";
    static final String unblockGroup = "unblockGroup";
    static final String updateGroupOwner = "updateGroupOwner";
    static final String addAdmin = "addAdmin";
    static final String removeAdmin = "removeAdmin";
    static final String muteMembers = "muteMembers";
    static final String unMuteMembers = "unMuteMembers";
    static final String muteAllMembers = "muteAllMembers";
    static final String unMuteAllMembers = "unMuteAllMembers";
    static final String addWhiteList = "addWhiteList";
    static final String removeWhiteList = "removeWhiteList";
    static final String uploadGroupSharedFile = "uploadGroupSharedFile";
    static final String downloadGroupSharedFile = "downloadGroupSharedFile";
    static final String removeGroupSharedFile = "removeGroupSharedFile";
    static final String updateGroupAnnouncement = "updateGroupAnnouncement";
    static final String updateGroupExt = "updateGroupExt";
    static final String joinPublicGroup = "joinPublicGroup";
    static final String requestToJoinPublicGroup = "requestToJoinPublicGroup";
    static final String acceptJoinApplication = "acceptJoinApplication";
    static final String declineJoinApplication = "declineJoinApplication";
    static final String acceptInvitationFromGroup = "acceptInvitationFromGroup";
    static final String declineInvitationFromGroup = "declineInvitationFromGroup";
    static final String setMemberAttributesFromGroup = "setMemberAttributesFromGroup";
    static final String removeMemberAttributesFromGroup = "removeMemberAttributesFromGroup";
    static final String fetchMemberAttributesFromGroup = "fetchMemberAttributesFromGroup";
    static final String fetchMembersAttributesFromGroup = "fetchMembersAttributesFromGroup";

    /// EMGroupManagerListener
    static final String onGroupChanged = "onGroupChanged";

    /// EMPushManager
    static final String getImPushConfig = "getImPushConfig";
    static final String getImPushConfigFromServer = "getImPushConfigFromServer";
    static final String enableOfflinePush = "enableOfflinePush";
    static final String disableOfflinePush = "disableOfflinePush";
    static final String updateImPushStyle = "updateImPushStyle";
    static final String updatePushNickname = "updatePushNickname";

    static final String updateGroupPushService = "updateGroupPushService";
    static final String getNoPushGroups = "getNoPushGroups";
    static final String updateUserPushService = "updateUserPushService";
    static final String getNoPushUsers = "getNoPushUsers";

    static final String updateHMSPushToken = "updateHMSPushToken";
    static final String updateFCMPushToken = "updateFCMPushToken";


    static final String reportPushAction = "reportPushAction";
    static final String setConversationSilentMode = "setConversationSilentMode";
    static final String removeConversationSilentMode = "removeConversationSilentMode";
    static final String fetchConversationSilentMode = "fetchConversationSilentMode";
    static final String setSilentModeForAll = "setSilentModeForAll";
    static final String fetchSilentModeForAll = "fetchSilentModeForAll";
    static final String fetchSilentModeForConversations = "fetchSilentModeForConversations";
    static final String setPreferredNotificationLanguage = "setPreferredNotificationLanguage";
    static final String fetchPreferredNotificationLanguage = "fetchPreferredNotificationLanguage";
    static final String setPushTemplate = "setPushTemplate";
    static final String getPushTemplate = "getPushTemplate";

    /// EMUserInfoManager 
    static final String updateOwnUserInfo = "updateOwnUserInfo";
    static final String updateOwnUserInfoWithType = "updateOwnUserInfoWithType";
    static final String fetchUserInfoById = "fetchUserInfoById";
    static final String fetchUserInfoByIdWithType = "fetchUserInfoByIdWithType";

    /// EMPresenceManager methods
    static final String presenceWithDescription = "publishPresenceWithDescription";
    static final String presenceSubscribe = "presenceSubscribe";
    static final String presenceUnsubscribe = "presenceUnsubscribe";
    static final String fetchSubscribedMembersWithPageNum = "fetchSubscribedMembersWithPageNum";
    static final String fetchPresenceStatus = "fetchPresenceStatus";

    /// EMPresenceManagerListener
    static final String onPresenceStatusChanged = "onPresenceStatusChanged";

    /// EMChatThreadManager methods
    static final String fetchChatThreadDetail = "fetchChatThreadDetail";
    static final String fetchJoinedChatThreads = "fetchJoinedChatThreads";
    static final String fetchChatThreadsWithParentId = "fetchChatThreadsWithParentId";
    static final String fetchJoinedChatThreadsWithParentId = "fetchJoinedChatThreadsWithParentId";
    static final String fetchChatThreadMember = "fetchChatThreadMember";
    static final String fetchLastMessageWithChatThreads = "fetchLastMessageWithChatThreads";
    static final String removeMemberFromChatThread = "removeMemberFromChatThread";
    static final String updateChatThreadSubject = "updateChatThreadSubject";
    static final String createChatThread = "createChatThread";
    static final String joinChatThread = "joinChatThread";
    static final String leaveChatThread = "leaveChatThread";
    static final String destroyChatThread = "destroyChatThread";

    // EMChatThreadManagerListener
    static final String onChatThreadCreate = "onChatThreadCreate";
    static final String onChatThreadUpdate = "onChatThreadUpdate";
    static final String onChatThreadDestroy = "onChatThreadDestroy";
    static final String onUserKickOutOfChatThread = "onUserKickOutOfChatThread";


    /// HandleAction
    static final String startCallback = "startCallback";
}
